### PR TITLE
add attention cells.

### PR DESCRIPTION
**Pull request by piiswrong**
dot attention

### CODE REVIEW

**Comment by piiswrong** [comment]
@tdomhan @mkolod

**Comment by eric-haibin-lin** [comment]
@szha

File: tests/python/unittest/test_rnn.py [hunk 1]
```diff
@@ -107,7 +107,30 @@ def test_unfuse():
     assert outs == [(10, 200), (10, 200), (10, 200)]
 
 
+def test_attention():
```
Comment (piiswrong) [inline tests/python/unittest/test_rnn.py]: @tdomhan

**Comment by szha** [comment]
LGTM

**Comment by piiswrong** [comment]
closed per discussion with NLP folks. A separate interface from RNNCell may make more sense.

Will propose a different design later.

@tdomhan

**Pull request closed without merging**